crates/starknet-devnet/tests/test_v3_transactions.rs

```diff
@@ -0,0 +1,472 @@
+    }
+
+    fn get_gas_units_and_gas_price(fee_estimate: FeeEstimate) -> (u64, u128) {
+        let gas_price =
+            u128::from_le_bytes(fee_estimate.gas_price.to_bytes_le()[0..16].try_into().unwrap());
+        let gas_units = fee_estimate
+            .overall_fee
+            .field_div(&NonZeroFelt::from_felt_unchecked(fee_estimate.gas_price));
+
+        (gas_units.to_le_digits().first().cloned().unwrap(), gas_price)
+    }
+    async fn transaction_with_less_gas_units_and_or_less_gas_price_should_return_error_or_be_accepted_as_reverted<
+        A: ConnectedAccount + Sync,
+        F: AccountFactory + Sync,
+    >(
+        transaction_action: Action,
+        account: Option<&A>,
+        account_factory: Option<&F>,
+    ) {
+        let estimate_fee = match &transaction_action {
+            Action::Declaration(sierra_class, casm_hash) => {
+                DeclarationV3::new(sierra_class.clone(), *casm_hash, account.unwrap())
+                    .estimate_fee()
+                    .await
+                    .unwrap()
+            }
+            Action::AccountDeployment(salt) => {
+                AccountDeploymentV3::new(*salt, account_factory.unwrap())
+                    .estimate_fee()
+                    .await
+                    .unwrap()
+            }
+            Action::Execution(calls) => {
+                ExecutionV3::new(calls.clone(), account.unwrap()).estimate_fee().await.unwrap()
+            }
+        };
+
+        let (estimated_gas_units, gas_price) = get_gas_units_and_gas_price(estimate_fee);
+
+        for (gas_units, gas_price) in [
+            (Some(estimated_gas_units - 1), Some(gas_price)),
+            (Some(estimated_gas_units), Some(gas_price - 1)),
+            (Some(estimated_gas_units - 1), Some(gas_price - 1)),
+            (Some(estimated_gas_units - 1), None),
+            (None, Some(gas_price - 1)),
+        ] {
+            match &transaction_action {
+                Action::Declaration(sierra_class, casm_hash) => {
+                    let mut declaration =
+                        DeclarationV3::new(sierra_class.clone(), *casm_hash, account.unwrap());
+
+                    if let Some(gas_units) = gas_units {
+                        declaration = declaration.gas(gas_units);
+                    }
+                    if let Some(gas_price) = gas_price {
+                        declaration = declaration.gas_price(gas_price);
+                    }
+                    match declaration.send().await.unwrap_err() {
+                        starknet_rs_accounts::AccountError::Provider(
+                            ProviderError::StarknetError(StarknetError::InsufficientMaxFee),
+                        ) => {}
+                        starknet_rs_accounts::AccountError::Provider(ProviderError::Other(
+                            other,
+                        )) if other.to_string().contains("is lower than the actual gas price") => {}
```

Comment:
Doesn't the fact that we have to match it via the string content of `ProviderError::Other` mean that Devnet is returning a wrong error? Or does starknet-rs just not know about the correct error type? Or is it one of those Devnet-specific situations where it doesn't need to conform to the spec?

Reply:
in the spec there is no error entry for errors of type Max L1 gas price(x) is lower than the actual gas price y. Non reverting transactions reject the transaction with this error message, but reverting transactions are accepted with the following revert reason

Reply:
I dont know if it can become InsufficientMaxFee